src/main/java/org/wildfly/security/auth/server/AbstractMechanismAuthenticationFactory.java

```diff
@@ -39,7 +39,7 @@
         this.mechanismConfigurations = mechanismConfigurations;
     }
 
-    SecurityDomain getSecurityDomain() {
+    public SecurityDomain getSecurityDomain() {
```

Comment:
Changing visibility on the parent class is sufficient for references to HttpAuthenticationFactory and SaslAuthenticationFactory to have access but I will move it to the interface as well.